Foundation/Process.swift

```diff
@@ -162,48 +162,36 @@ open class Process: NSObject {
     //
     
     public override init() {
-    
+
     }
-    
-    // These methods can only be set before a launch.
-    
-    open var launchPath: String?
+
+    // These properties can only be set before a launch.
+    open var executableURL: URL?
+    open var currentDirectoryURL = URL(fileURLWithPath: FileManager.default.currentDirectoryPath, isDirectory: true)
     open var arguments: [String]?
     open var environment: [String : String]? // if not set, use current
-    
-    open var currentDirectoryPath: String = FileManager.default.currentDirectoryPath
-    
-    open var executableURL: URL? {
-        get {
-            guard let launchPath = self.launchPath else {
-                return nil
-            }
-            
-            return URL(fileURLWithPath: launchPath)
-        }
-        set {
-            self.launchPath = newValue?.path
-        }
+
+    @available(*, deprecated: 4, renamed: "executableURL")
```

Comment:
While this is allowed, I don't think the semantics for this are well-defined. (The '4' in the `deprecated:` portion is meant to be the version of the platform specified just after `@available(`, but in this case there's none.)

The semantically appropriate thing to do here, I think, is just to mark it:

```swift
@available(*, deprecated, renamed:…)
```

Reply:
Unless it is undocumented that the Swift version is used even if the tag is not `@available(swift, …)`

Reply:
I wasn't sure if a version number was required, its fixed now.